librz/util/subprocess.c

```diff
@@ -132,29 +136,29 @@ static LPWCH override_env(const char *envvars[], const char *envvals[], size_t e
 	return ret;
 }
 
-static void remove_cr(char *str) {
-	char *start = str;
-	while (*str) {
-		if (str[0] == '\r' &&
-			!(str - start >= 4 && !strncmp(str - 4, RZ_CONS_CLEAR_SCREEN, 4))) {
-			memmove(str, str + 1, strlen(str + 1) + 1);
-			continue;
-		}
-		str++;
-	}
-}
-
```

Comment:
@wargio I see this was removed from subprocess, but now old users of the subprocess API has to do this themselves. Did you remove this because you want to get the raw stdout/stderr of a program?

What do you think if we use these API?
```
RZ_API char *rz_subprocess_out_str(RzSubprocess *proc);
RZ_API char *rz_subprocess_err_str(RzSubprocess *proc);
RZ_API ut8 *rz_subprocess_out(RzSubprocess *proc, int *length);
RZ_API ut8 *rz_subprocess_err(RzSubprocess *proc, int *length);
```
The `_str` APIs apply `remove_cr` before returning the strings, so we don't need to replicate this function everywhere.

Together with this change we should also change a bit the `RzSubprocessOutput` structure. Something like:
```
typedef struct rz_process_output_t {
	char *out_str; // stdout
	char *err_str; // stderr
        ut8 *out, *err;
        int out_len, err_len;
	int ret; // exit code of the process
	bool timeout;
} RzSubprocessOutput;
```
Comments?

Reply:
stdout needs to be raw. it's up to the user to know what there will be in the buffer.
this is the reason you can do `cat mybinfile | hexdump -C` for example

Reply:
The remove_cr function shall not be there because the only one complaining about this is the testsuite, therefore that is a testsuite issue not an issue related to the pipe, nor subprocess

Reply:
I agree, that's why I kept the raw stdout in the proposal above. However, other times you just want the string and you want a "cleaned" string (e.g. by running remove_cr function), so I was thinking about providing that from RzSubprocess itself instead of having every user of the API definiting their own remove_cr function.

Also, no matter what we should probably add the `out_len`/`err_len` fields.

Reply:
cc @thestr4ng3r you wrote that piece of code, right?

Reply:
This `remove_cr` implementation is super inefficient so I can only assume I wrote it to only be used as part of `rz-test`, where it doesn't really matter.
If another user of `RzSubprocess` needs this, the functionality should not be provided as part of the subprocess api but as just an independent `rz_str_` function that can be called.
So for me the move to `rz-test/run.c` as it is done here seems good.

Reply:
Ok thanks for confirming @thestr4ng3r . Then I will just make sure to add `out_len`/`err_len` to `RzSubprocessOutput` and that's it.